pybamm/models/submodels/electrolyte_diffusion/composite_diffusion.py

```diff
@@ -75,53 +77,31 @@ def set_rhs(self, variables):
 
     def _get_source_terms_leading_order(self, variables):
         param = self.param
-        c_e_n = variables["Negative electrolyte concentration"]
-        c_e_p = variables["Positive electrolyte concentration"]
-
-        return sum(
-            pybamm.Concatenation(
-                (reaction["Negative"]["s"] - param.t_plus(c_e_n))
-                * variables["Leading-order " + reaction["Negative"]["aj"].lower()],
-                pybamm.FullBroadcast(0, "separator", "current collector"),
-                (reaction["Positive"]["s"] - param.t_plus(c_e_p))
-                * variables["Leading-order " + reaction["Positive"]["aj"].lower()],
-            )
-            / self.param.gamma_e
-            for reaction in self.reactions.values()
-        )
+
+        # All possible reactions. Some of these could be zero
+        j = variables["Leading-order interfacial current density"]
+        j_ox = variables["Leading-order oxygen interfacial current density"]
+
+        return (-param.s_plus_S * j - param.s_plus_Ox * j_ox) / self.param.gamma_e
 
     def _get_source_terms_first_order(self, variables):
         param = self.param
-        c_e_n = variables["Negative electrolyte concentration"]
-        c_e_p = variables["Positive electrolyte concentration"]
-
-        return sum(
-            pybamm.Concatenation(
-                (reaction["Negative"]["s"] - param.t_plus(c_e_n))
-                * variables[reaction["Negative"]["aj"]],
-                pybamm.FullBroadcast(0, "separator", "current collector"),
-                (reaction["Positive"]["s"] - param.t_plus(c_e_p))
-                * variables[reaction["Positive"]["aj"]],
-            )
-            / self.param.gamma_e
-            for reaction in self.reactions.values()
-        )
+
+        # All possible reactions. Some of these could be zero
+        j = variables["Interfacial current density"]
+        j_ox = variables["Oxygen interfacial current density"]
+
+        return (-param.s_plus_S * j - param.s_plus_Ox * j_ox) / self.param.gamma_e
 
     def _get_source_terms_first_order_average(self, variables):
-        first_order_average = sum(
-            (
-                reaction["Negative"]["s"]
-                * variables[
-                    "First-order x-averaged " + reaction["Negative"]["aj"].lower()
-                ]
-                + reaction["Positive"]["s"]
-                * variables[
-                    "First-order x-averaged " + reaction["Positive"]["aj"].lower()
-                ]
-            )
-            / self.param.gamma_e
-            for reaction in self.reactions.values()
-        )
+        param = self.param
+        # All possible reactions. Some of these could be zero
+        j_av = variables["First-order x-averaged interfacial current density"]
```

Comment:
it seems like there are a lot of places where you need to remember to add a new 'j'. Could we pass the list of reaction names in and then search for the side reactions that need to be added on? e.g. look for "First-order x-averaged" + "reaction_name" + "interfacial current density". Maybe there are some quirks that make this hard, but I'm worried people will forget to add the extra current densities

Reply:
I guess then you need things like s_plus values. And then we basically end up back at the reaction dictionary... I don't think I appreciated how often you needed to get each of the current densities in separate files

Reply:
Yeah exactly ... I'm leaning towards having to do everything explicitly though. I'll resolve conflicts and fix tests for now, and we can discuss at next meeting

Reply:
Maybe having "Total negative electrolyte reactions", "Total negative solid reactions", etc variables in the variables dictionary. The electrolyte one can include the s_plus values. When you add a new reaction, you then just do: 
```
variables["Total electrolyte reactions"] += s_plus * j_new
```
If the reaction is set to `NoReaction` then we just don't update it. If the reaction only consumes a reactant from the electrolyte, then we only add it to the electrolyte and not the solid. 

This would be a bit cleaner than the current reactions dictionary and also would make it clearer from just looking at a particular reaction submodel where that reaction fits into the wider model. Also good is that you only have to edit that one reaction submodel to control how that reaction is coupled to the full model. Downside is that it is less clear than the approach in this issue to see what reactions are taking place just by looking at the electrolyte submodel.

Reply:
Yeah, I think that's the cleanest way of doing it, then it just goes into `BaseInterface` and all happens automatically